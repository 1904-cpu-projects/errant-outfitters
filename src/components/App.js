import React from "react";

class App extends React.Component {
  // gotta see that HR!
  render() {
    return (
      <div>
        <hr />
      </div>
    );
  }
}

export default App;
