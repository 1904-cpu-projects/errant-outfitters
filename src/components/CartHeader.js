import React from 'react';
import { Link } from 'react-router-dom';

// numberItems needs to just be a number
// pull in from redux store CartStore I think
export function CartHeader({ cart }) {
  const cartItems = cart ? cart.length : 0;
  return (
    <div>
      <Link to="/user/profile">Cart</Link> | {cartItems}
    </div>
  );
}
