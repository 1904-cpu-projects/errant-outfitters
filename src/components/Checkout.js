import React from 'react';
import PropTypes from 'prop-types';
import { Elements, StripeProvider } from 'react-stripe-elements';
import { connect } from 'react-redux';

import StripCard from './StripeCard';
import { getCart } from '../storeReducers/cartReducer';
import { listProductsThunk } from '../actions/productActions';

class Checkout extends React.Component {
  constructor() {
    super();
    this.state = {
      total: 0,
      transactionComplete: false,
      transacitons: [],
    };
    this.updateCart = this.updateCart.bind(this);
  }

  calcTotal(cart) {
    const total = cart.items.reduce((acc, item) => {
      return acc + item.product.cost * item.quantity;
    }, 0);
    this.setState({ total: total });
  }

  updateCart(data) {
    console.log(data);
    this.props.getCart();
    this.props.listProductsThunk();
    this.setState({ transactionComplete: true, transactions: [...data] });
  }

  componentDidMount() {
    const { cart } = this.props;
    this.calcTotal(cart);
  }

  render() {
    if (!this.state.transactionComplete) {
      return (
        <div className="checkout">
          <p>Would you like to complete the purchase?</p>
          <div>
            <StripeProvider apiKey={process.env.STRIPE_API}>
              <Elements>
                <StripCard updateCart={this.updateCart} />
              </Elements>
            </StripeProvider>
          </div>
          <div>
            <h2>Your total for this purchase is ${this.state.total}</h2>
            <p>
              You don&apos;t even need to worry about shipping information
              <br />
              We have mages on standby to teleport your items directly to you!
            </p>
          </div>
        </div>
      );
    } else {
      return (
        <div>
          <h2>Your transaction is complete!</h2>
          <div>
            Total cost: {this.state.transactions[0].totalCost}
            <br />
            Transaction #: {this.state.transactions[0].id}
            <br />
            Your Items:
            <br />
            {this.state.transactions.map((item, idx) => (
              <p key={idx}>
                {item.product.name} | Quantity: {item.quantity}
              </p>
            ))}
          </div>
        </div>
      );
    }
  }
}

Checkout.propTypes = {
  cart: PropTypes.object,
  getCart: PropTypes.func,
  listProductsThunk: PropTypes.func,
};

const mapStateToProps = state => ({
  cart: state.cart,
});

const mapDispatchToProps = dispatch => ({
  getCart: () => dispatch(getCart()),
  listProductsThunk: () => dispatch(listProductsThunk()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Checkout);
