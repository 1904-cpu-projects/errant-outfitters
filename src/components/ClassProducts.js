import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { SingleProduct } from './SingleProduct';

export function Products({ products, user }) {
  const classProducts = products.productList.filter(
    a => a.class === user.class,
  );
  //function to make tiles from SingleProduct component
  if (classProducts.length !== 0) {
    return (
      <div>
        <div className="products-list">
          {classProducts.map(p => {
            return <SingleProduct key={p.id} product={p} />;
          })}
        </div>
      </div>
    );
  } else {
    return (
      <div>
        <h2 className="products-list">
          Please register/login to see your class recommended products
        </h2>
      </div>
    );
  }
}

Products.propTypes = {
  productList: PropTypes.array,
  user: PropTypes.object,
};

const mapStateToProps = state => ({
  products: state.products,
  user: state.user,
});

export default connect(mapStateToProps)(Products);
