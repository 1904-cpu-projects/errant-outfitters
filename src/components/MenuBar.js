import React from 'react';
import { NavLink } from 'react-router-dom';
import { CartHeader } from './CartHeader';

function MenuBar(cart) {
  return (
    <div id="menu-bar">
      <div>
        <NavLink className="navlink" exact to="/">
          <button>All Products</button>
        </NavLink>
        <NavLink className="navlink" exact to="/armor">
          <button>Armor</button>
        </NavLink>
        <NavLink className="navlink" exact to="/weapon">
          <button>Weapons</button>
        </NavLink>
        <NavLink className="navlink" exact to="/potion">
          <button>Potions</button>
        </NavLink>
        <NavLink className="navlink" exact to="/ClassProducts">
          <button>Class Products</button>
        </NavLink>
      </div>
      <div>
        <NavLink className="navlink" exact to="/costA">
          <button>Cost Asc</button>
        </NavLink>
        <NavLink className="navlink" exact to="/costD">
          <button>Cost Desc</button>
        </NavLink>
        <NavLink className="navlink" exact to="/inStock">
          <button>In Stock</button>
        </NavLink>
        <NavLink className="navlink" exact to="/outStock">
          <button>Out of Stock</button>
        </NavLink>
        <NavLink className="navlink" exact to="/stockA">
          <button>Stock Asc</button>
        </NavLink>
        <NavLink className="navlink" exact to="/stockD">
          <button>Stock Desc</button>
        </NavLink>
      </div>
      <div>
        <form method="get" action="/auth/google">
          <button type="submit" className="btn bg-red white p1 rounded">
            Login with Google
          </button>
        </form>
        <CartHeader cart={cart} />
      </div>
    </div>
  );
}

export default MenuBar;
