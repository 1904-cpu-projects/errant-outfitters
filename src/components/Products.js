import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import SingleProduct from './SingleProduct';
import ClassProducts from './ClassProducts';

export function Products({ products, match }) {
  let displayFilter = [];
  if (match.params.filterBar === 'ClassProducts') return <ClassProducts />;
  match.params.filterBar === 'inStock'
    ? (displayFilter = products.productList.filter(p => p.inStock))
    : match.params.filterBar === 'outStock'
    ? (displayFilter = products.productList.filter(p => !p.inStock))
    : (displayFilter = match.params.filterBar
        ? products.productList.filter(
            p => p.catagory === match.params.filterBar,
          )
        : products.productList);

  //function to make tiles from SingleProduct component
  return (
    <div className="products-list">
      {displayFilter.map(p => {
        return <SingleProduct key={p.id} product={p} />;
      })}
    </div>
  );
}

Products.defaultProps = {
  productList: [],
  products: [],
  match: {},
};

Products.propTypes = {
  products: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
    PropTypes.number,
  ]),
  productList: PropTypes.arrayOf(PropTypes.object),
  match: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.number,
  ]),
};

const mapStateToProps = state => ({
  products: state.products,
});

export default connect(mapStateToProps)(Products);
