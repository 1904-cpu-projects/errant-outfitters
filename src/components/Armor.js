import React from 'react';
import { connect } from 'react-redux';
import { SingleProduct } from './SingleProduct';
import MenuBar from './MenuBar';

export function Products({ products }) {
  const armor = products.productList.filter(a => a.catagory === 'armor');
  //function to make tiles from SingleProduct component

  return (
    <div>
      <MenuBar />
      <div className="products-list">
        {armor.map(p => {
          return <SingleProduct key={p.id} product={p} />;
        })}
      </div>
    </div>
  );
}

const mapStateToProps = state => ({
  products: state.products,
});

export default connect(mapStateToProps)(Products);
