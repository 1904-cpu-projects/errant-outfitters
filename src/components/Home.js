import React from 'react';

import Products from './Products';
import MenuBar from './MenuBar';

function Home() {
  return (
    <div className="main-view">
      <MenuBar />
      <Products />
    </div>
  );
}
