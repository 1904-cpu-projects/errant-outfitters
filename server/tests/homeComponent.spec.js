import React from "react";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallow, mount, render } from "enzyme";
import { Home } from "../../src/components/Home";
import {MenuBar} from "../../src/components/MenuBar";
import {Products} from '../../src/components/Products'

configure({ adapter: new Adapter() });

describe("Correctly renders main components component", () => {
  it("renders the home component present", () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(MenuBar)).to.have.lengthOf(1)
  });
});
