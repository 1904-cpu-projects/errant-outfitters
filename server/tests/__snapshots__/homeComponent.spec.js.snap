// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly renders main components component renders the home component present 1`] = `ShallowWrapper {}`;
